assets/javascripts/discourse/initializers/algoliasearch.js

```diff
@@ -0,0 +1,2380 @@
+/* eslint-disable */
+export default {
+  name: "algoliasearch",
+
+  initialize() {
```

Comment:
Can we defer the loading of this script until the feature is required? We have a `loadScript` pattern which we use in core:  https://github.com/discourse/discourse/search?q=loadScript